### PR TITLE
Orca: Remove the loop of fit in our example Super Resolution

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

Call estimator.fit in a loop may raise a big overhead when there is a short period of time for real training, and it seems that our examples are misleading our user to do that.
https://github.com/analytics-zoo/Telecom-CTC-E2E/issues/27#issue-1527156829

So we just apply another way to achieve this.

### CODE REVIEW

File: python/orca/example/learn/pytorch/super_resolution/super_resolution.py [hunk 1]
```diff
@@ -302,21 +303,27 @@ def optim_creator(model, config):
...
+
+    val_stats = estimator.evaluate(data=validation_data_creator,
+                                    batch_size=opt.test_batch_size)
+    print("===> Validation Complete: Avg. PSNR: {:.4f} dB, Avg. Loss: {:.4f}"
+            .format(10 * log10(1. / val_stats["val_loss"]), val_stats["val_loss"]))
+
+    
+    model = estimator.get_model()
+    last_model_path = f_model_out_path.format(epoch=opt.epoch)
+    torch.save(model, last_model_path)
```
Comment (hkvision) [inline python/orca/example/learn/pytorch/super_resolution/super_resolution.py:325]: Can we change to estimator.save?

File: python/orca/example/learn/pytorch/super_resolution/super_resolution.py [hunk 2]
```diff
@@ -302,21 +303,27 @@ def optim_creator(model, config):
...
-        model_out_path = model_dir + "/" + "model_epoch_{}.pth".format(epoch)
-        model = estimator.get_model()
-        torch.save(model, model_out_path)
-        print("Checkpoint saved to {}".format(model_out_path))
+    f_model_out_path = model_dir + "/" + "model_epoch_{epoch:02d}.pth"
+
+    stats = estimator.fit(data=train_data_creator,
+                          epochs=opt.epochs,
+                          batch_size=opt.batch_size,
+                          callbacks=[ModelCheckpoint(filepath=f_model_out_path,
```
Comment (hkvision) [inline python/orca/example/learn/pytorch/super_resolution/super_resolution.py:311]: does {epoch:02d} work to input here?

File: python/orca/example/learn/pytorch/super_resolution/super_resolution.py [hunk 3]
```diff
@@ -299,24 +301,29 @@ def optim_creator(model, config):
...
-        makedirs(model_dir)
+    if exists(model_dir):
+        shutil.rmtree(model_dir)
 
-    for epoch in range(1, opt.epochs + 1):
-        stats = estimator.fit(data=train_data_creator, epochs=1, batch_size=opt.batch_size)
-        for epochinfo in stats:
-            print("===> Epoch {} Complete: Avg. Loss: {:.4f}"
-                  .format(epoch, epochinfo["train_loss"]))
+    f_model_out_path = model_dir + "/" + "model_epoch_{epoch:02d}.pth"
```
Comment (hkvision) [inline python/orca/example/learn/pytorch/super_resolution/super_resolution.py:307]: use os.path.join?

File: python/orca/example/learn/pytorch/super_resolution/super_resolution.py [hunk 4]
```diff
@@ -299,24 +301,29 @@ def optim_creator(model, config):
         }
     )
 
-    if not exists(model_dir):
-        makedirs(model_dir)
+    if exists(model_dir):
+        shutil.rmtree(model_dir)
```
Comment (hkvision) [inline python/orca/example/learn/pytorch/super_resolution/super_resolution.py:305]: Does this work for hdfs? Or we only save the model to driver in this example?

Comment (leonardozcm) [reply python/orca/example/learn/pytorch/super_resolution/super_resolution.py:305]: > Does this work for hdfs? Or we only save the model to driver in this example?

I believe this example do not involve hdfs

**Pull request merged**